### PR TITLE
Include Parameters and Annotations when Copying an Action

### PR DESCRIPTION
**Pull request by dubeejw**
- When copying an action ensure that parameters and annotations are also copied

Closes https://github.com/openwhisk/openwhisk/issues/418

### CODE REVIEW

**Comment by dubeejw** [comment]
PG2 8​24

**Comment by dubeejw** [comment]
@rabbah, any comments on this one?

**Comment by rabbah** [comment]
I'm having second thoughts about this - may be a good one to take on the dev list.
The reason: we don't make it possible to delete a parameter from the CLI.

**Comment by csantanapr** [comment]
@rabbah I think this is good thing to add to the CLI, be able to copy all the metadata.

The reason: we don't make it possible to delete a parameter from the CLI.
What's the concern here about deleting a parameter?

Merging this PR doesn't block us from offering a way to delete a single parameter in future update of CLI.

File: tools/cli/go-whisk-cli/commands/action.go [hunk 1]
```diff
@@ -557,7 +559,9 @@ func parseAction(cmd *cobra.Command, args []string) (*whisk.Action, error) {
 
   action.Name = qName.entityName
   action.Namespace = qName.namespace
-  action.Annotations = annotations.(whisk.KeyValueArr)
+  if !flags.action.copy {
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/commands/action.go]: What does this guard?

File: tests/src/system/basic/WskActionTests.scala [hunk 2]
```diff
@@ -134,6 +134,34 @@ class WskActionTests
...
+            }
+
+            val originActionAnnots = wsk.action.get(origActionName, fieldFilter = Some("annotations")).stdout
+            val copiedActionAnnots = wsk.action.get(copiedActionName, fieldFilter = Some("annotations")).stdout
+            val originActionParams = wsk.action.get(origActionName, fieldFilter = Some("parameters")).stdout
+            val copiedActionParams = wsk.action.get(copiedActionName, fieldFilter = Some("parameters")).stdout
+            val originActionExec = wsk.action.get(origActionName, fieldFilter = Some("exec")).stdout
+            val copiedActionExec = wsk.action.get(copiedActionName, fieldFilter = Some("exec")).stdout
+
+            copiedActionAnnots.substring(copiedActionAnnots.indexOf("\n")) shouldBe originActionAnnots.substring(originActionAnnots.indexOf("\n"))
```
Comment (rabbah) [inline tests/src/system/basic/WskActionTests.scala]: Should check that the semver of the copy is 0.0.1 too?

File: tests/src/system/basic/WskActionTests.scala [hunk 3]
```diff
@@ -134,6 +134,34 @@ class WskActionTests
...
+            copiedActionParams.substring(copiedActionParams.indexOf("\n")) shouldBe originActionParams.substring(originActionParams.indexOf("\n"))
```
Comment (rabbah) [inline tests/src/system/basic/WskActionTests.scala]: Is the substring necessary?

Comment (dubeejw) [inline tests/src/system/basic/WskActionTests.scala]: I updated the test to make it more readable, and added some methods that will make writing tests that require examination of non-activation record responses from the CLI easier in the future.

**Comment by dubeejw** [comment]
PG2 1029

**Pull request merged**